### ISSUE TITLE
afterDelete receives `[result]` not `result` when deleting from the collection page

### ISSUE THREAD

**Issue by laurenskling** [strapi/strapi#8317]
**Describe the bug**
Deleting items from the overview (by selecting them and pressing the "delete selected" button) will result in the `afterDelete` lifecycle receiving an Array with the deleted item, not just the deleted item.

**Steps to reproduce the behavior**
0. Create a model with a `afterDelete` lifecycle.
1. Go to a collecting, like: http://0.0.0.0:1337/admin/plugins/content-manager/collectionType/plugins::pages.pages
2. Select 1, 2 or all items.
3. Press the "deleted selected" (or "delete all") button
4. every entry will now call the afterDelete lifecycle
5. see that the first param (`result`) is now an array containing the result. Not just the result itself

**Expected behavior**
It's not 1 afterDelete being called with an array of all deleted entries as well. Its all entries, hitting the lifecycle all themselfs, all with an array.length === 1

I'd expect afterDelete always to just receive the `result`

**System**

- Node.js version: 12.18
- NPM version: 6
- Strapi version: 3.2.0
- Database: mongo
- Operating system: macos

**Additional context**
link to slack message about it: https://strapi.slack.com/archives/C0BNGCDNH/p1602506645051400
@MattieBelt found this code that might be relevant: https://github.com/strapi/strapi/blob/94154d31dfd4efabf73729cc86f2fb58ff50429e/packages/strapi-connector-mongoose/lib/queries.js#L528

**Comment by derrickmehaffy** [comment]
Tested in v3.2.3 via SQLite results in an array yes, but does return it in a way that is still useful (the lifecycle code will need to adjust and check if `result` is an object or an array):

![image](https://user-images.githubusercontent.com/8593673/95911209-08bec280-0d56-11eb-99d7-fdea9bdde556.png)

The same applies to Mongo:

![image](https://user-images.githubusercontent.com/8593673/95911409-50dde500-0d56-11eb-8082-891a02334812.png)


While this is "unexpected" if you are only selecting one entry within the CTM and using the bulk delete function, it's not actually causing an error nor is it really a bug per say. I think we just need to add documentation for it.

**Comment by derrickmehaffy** [comment]
(I do see where an internal loop triggering multiple `before/after Delete` actions could be favorable, in terms of code quality an execution time, it may be better to handle this as a loop within the lifecycle)

I'll poke @alexandrebodin to get his thoughts though, and allow him to make the final judgement.

**Comment by laurenskling** [comment]
It's not a crash, but it's confusing and not in documentation. So we should fix _something_. Ofcourse I would prefer fixing it not being an array, because it doesn't make any sense that it's an array right now; and you would have to write handlers everywhere for it, because of how the UI works. I would be happy to help to fix this.

**Comment by alexandrebodin** [comment]
Hi, Currently the delete method handles both single and multiple deletes that's why it will return one or both. Without getting into too much details we can't do that looping as @derrickmehaffy suggested for now. We will keep this into our backlog for the incoming work on the db layer as it is not a blocker.

**Comment by laurenskling** [comment]
One would think the lifecycle of a model wouldn't have to do anything with methods allowing multiple deletes, right? When you delete multiple items, the lifecycles will be called _for every model_ with an array with only that current model inside. So it doesn't make sense. 

I understand more important things could be on your plate. But let's keep this open, I might try to fix it.

**Comment by alexandrebodin** [comment]
Yes let's keep it open. I already have added it into the tech backlog for the DB layer

**Comment by Crazysipu** [comment]
> Gonna close this in favor of #12501 as we now do have a deleteMany query (and lifecycle)

how to solve this problem??

**Comment by derrickmehaffy** [comment]
> > Gonna close this in favor of #12501 as we now do have a deleteMany query (and lifecycle)
> 
> how to solve this problem??

Because there are two entirely different lifecycle actions. One for regular delete and one for bulk
